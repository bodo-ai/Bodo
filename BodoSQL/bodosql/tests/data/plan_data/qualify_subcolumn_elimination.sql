select
    A,
    C,
    E,
    MEDIAN(A)
        OVER (
            PARTITION BY C ORDER BY E ASC ROWS BETWEEN UNBOUNDED PRECEDING AND 1 PRECEDING
        ) as WINDOW_AGG_ASC
FROM table1
QUALIFY
    WINDOW_AGG_ASC > 1 OR MEDIAN(A) OVER (
        PARTITION BY C ORDER BY E DESC ROWS BETWEEN UNBOUNDED PRECEDING AND 1 PRECEDING) >= 1
ORDER BY C, E
