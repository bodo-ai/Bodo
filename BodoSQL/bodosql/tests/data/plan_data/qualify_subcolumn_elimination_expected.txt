LogicalSort(sort0=[$1], sort1=[$2], dir0=[ASC-nulls-first], dir1=[ASC-nulls-first])
  LogicalProject(A=[$0], C=[$1], E=[$2], WINDOW_AGG_ASC=[$3])
    LogicalFilter(condition=[$4])
      LogicalProject(A=[$0], C=[$1], E=[$2], WINDOW_AGG_ASC=[$3], EXPR$4=[OR(>($3, 1), >=(MEDIAN($4) OVER (PARTITION BY $5 ORDER BY $6 DESC RANGE BETWEEN UNBOUNDED PRECEDING AND 1 PRECEDING), 1))])
        LogicalProject(A=[$0], C=[$2], E=[$4], WINDOW_AGG_ASC=[MEDIAN($0) OVER (PARTITION BY $2 ORDER BY $4 ROWS BETWEEN UNBOUNDED PRECEDING AND 1 PRECEDING)], $f4=[$0], $f5=[$2], $f6=[$4])
          LogicalTableScan(table=[[__bodolocal__, table1]])
